client/src/lib.rs

```diff
@@ -114,6 +114,11 @@ pub struct ClientOptions {
     /// Retry configuration for the server client. Default is [RetryConfig::default]
     #[builder(default)]
     pub retry_config: RetryConfig,
+
+    /// If set, override the origin used when connecting. May be useful in conjunction with
+    /// [TlsConfig::domain] when routing connections through some form of proxy.
+    #[builder(default)]
+    pub override_origin: Option<Uri>,
```

Comment:
Often this would be just overriding authority (e.g. https://pkg.go.dev/google.golang.org/grpc#WithAuthority) and would not have to be a full URL with scheme and path and port and such. On lang side we hide the full client URL that Tonic uses in favor of a simple host:port. I wonder how this value will look lang side.

Reply:
Yeah, I can't do that unfortunately. I've cleaned it up so you don't need to specify this and we'll just construct it from the domain override.

Tonic doesn't let me set the authority header directly which is... annoying. I'll see about opening an item there for this.